### PR TITLE
Add key constraints for the TPC-DS benchmark

### PR DESCRIPTION
**Pull request by julianmenzler**
Adds all PRIMARY KEY key constraints from the TPC-DS spec, so that we can profit from unique constraints and FDs in our query plans.

**TPC-DS, SF=1**
<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------------------------------------------+------------------------------------------------+
 | Parameter        | results_master.json                            | results_constraints.json                       |
 +------------------+------------------------------------------------+------------------------------------------------+
 |  GIT-HASH        | 2905a7817d9d8d62581160431c07ff657e537441-dirty | c0e706da1384d8c0bfd54f8e5590c63b1956bc61-dirty |
 |  benchmark_mode  | Ordered                                        | Ordered                                        |
 |  build_type      | release                                        | release                                        |
 |  chunk_size      | 65535                                          | 65535                                          |
 |  clients         | 1                                              | 1                                              |
 |  compiler        | gcc 9.2                                        | gcc 9.2                                        |
 |  cores           | 0                                              | 0                                              |
 |  date            | 2020-10-02 16:43:45                            | 2020-10-02 18:06:49                            |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}        | {'default': {'encoding': 'Dictionary'}}        |
 |  indexes         | False                                          | False                                          |
 |  max_duration    | 60000000000                                    | 60000000000                                    |
 |  max_runs        | -1                                             | -1                                             |
 |  time_unit       | ns                                             | ns                                             |
 |  using_scheduler | False                                          | False                                          |
 |  verify          | False                                          | False                                          |
 |  warmup_duration | 0                                              | 0                                              |
 +------------------+------------------------------------------------+------------------------------------------------+
```
</details>

```diff
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |     new |        ||      old |      new |        |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | 01      ||     59.0 |    56.8 |   -4%  ||    16.95 |    17.59 |   +4%  |  0.0000 |
 | 03      ||     11.8 |    11.8 |   +0%  ||    84.90 |    84.69 |   -0%  |  0.0000 |
 | 06      ||     36.8 |    37.1 |   +1%  ||    27.14 |    26.99 |   -1%  |  0.0000 |
+| 07      ||    118.9 |    66.5 |  -44%  ||     8.41 |    15.04 |  +79%  |  0.0000 |
 | 09      ||    272.7 |   273.8 |   +0%  ||     3.67 |     3.65 |   -0%  |  0.0218 |
+| 10      ||     41.1 |    39.1 |   -5%  ||    24.34 |    25.54 |   +5%  |  0.0000 |
 | 13      ||    214.2 |   215.0 |   +0%  ||     4.67 |     4.65 |   -0%  |  0.8476 |
+| 15      ||     21.8 |    20.1 |   -8%  ||    45.78 |    49.67 |   +8%  |  0.0000 |
+| 17      ||     60.5 |    55.9 |   -8%  ||    16.52 |    17.89 |   +8%  |  0.0000 |
 | 19      ||     16.6 |    16.3 |   -2%  ||    60.25 |    61.17 |   +2%  |  0.0000 |
+| 25      ||     31.4 |    28.0 |  -11%  ||    31.84 |    35.71 |  +12%  |  0.0000 |
+| 26      ||     57.5 |    30.8 |  -46%  ||    17.38 |    32.43 |  +87%  |  0.0000 |
 | 28      ||   1694.7 |  1699.3 |   +0%  ||     0.59 |     0.59 |   -0%  |  0.0068 |
+| 29      ||     60.7 |    48.5 |  -20%  ||    16.46 |    20.62 |  +25%  |  0.0000 |
 | 31      ||    252.4 |   252.9 |   +0%  ||     3.96 |     3.95 |   -0%  |  0.4365 |
 | 34      ||     52.2 |    52.0 |   -0%  ||    19.14 |    19.24 |   +0%  |  0.0053 |
+| 35      ||    202.2 |   179.8 |  -11%  ||     4.94 |     5.56 |  +12%  |  0.0000 |
+| 39a     ||    390.1 |   360.8 |   -8%  ||     2.56 |     2.77 |   +8%  |  0.0000 |
+| 39b     ||    387.0 |   352.9 |   -9%  ||     2.58 |     2.83 |  +10%  |  0.0000 |
 | 41      ||     59.0 |    59.5 |   +1%  ||    16.96 |    16.82 |   -1%  |  0.0000 |
 | 42      ||     12.4 |    12.5 |   +1%  ||    80.45 |    79.94 |   -1%  |  0.0000 |
 | 43      ||    586.0 |   587.2 |   +0%  ||     1.71 |     1.70 |   -0%  |  0.0134 |
+| 45      ||     20.1 |    18.8 |   -6%  ||    49.81 |    53.13 |   +7%  |  0.0000 |
+| 48      ||    181.7 |   152.1 |  -16%  ||     5.50 |     6.58 |  +19%  |  0.0000 |
+| 50      ||     20.0 |    19.1 |   -5%  ||    49.89 |    52.40 |   +5%  |  0.0000 |
 | 52      ||     12.7 |    12.8 |   +1%  ||    78.87 |    78.32 |   -1%  |  0.0000 |
 | 55      ||     12.5 |    12.3 |   -2%  ||    79.71 |    81.27 |   +2%  |  0.0000 |
+| 62      ||    148.0 |   139.8 |   -6%  ||     6.76 |     7.15 |   +6%  |  0.0000 |
+| 65      ||    154.8 |   129.4 |  -16%  ||     6.46 |     7.73 |  +20%  |  0.0000 |
+| 69      ||     54.8 |    47.9 |  -13%  ||    18.24 |    20.87 |  +14%  |  0.0000 |
 | 73      ||     32.6 |    33.6 |   +3%  ||    30.67 |    29.73 |   -3%  |  0.0000 |
+| 79      ||    105.8 |    94.3 |  -11%  ||     9.45 |    10.61 |  +12%  |  0.0000 |
 | 81      ||     36.0 |    34.9 |   -3%  ||    27.75 |    28.63 |   +3%  |  0.0000 |
 | 83      ||     17.6 |    17.4 |   -1%  ||    56.81 |    57.45 |   +1%  |  0.0000 |
+| 85      ||    110.7 |   101.0 |   -9%  ||     9.03 |     9.90 |  +10%  |  0.0057 |
 | 88      ||    158.8 |   160.4 |   +1%  ||     6.30 |     6.24 |   -1%  |  0.0006 |
 | 91      ||     33.5 |    32.6 |   -3%  ||    29.83 |    30.68 |   +3%  |  0.0000 |
 | 93      ||    492.7 |   493.7 |   +0%  ||     2.03 |     2.03 |   -0%  |  0.2760 |
 | 96      ||     14.1 |    14.2 |   +1%  ||    70.97 |    70.55 |   -1%  |  0.0000 |
 | 97      ||    840.4 |   847.2 |   +1%  ||     1.19 |     1.18 |   -1%  |  0.0223 |
+| 99      ||    305.4 |   285.0 |   -7%  ||     3.27 |     3.51 |   +7%  |  0.0000 |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Sum     ||   7391.3 |  7103.0 |   -4%  ||          |          |        |         |
+| Geomean ||          |         |        ||          |          |   +8%  |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
```

### CODE REVIEW

File: src/benchmarklib/tpcds/tpcds_table_generator.hpp [hunk 1]
```diff
@@ -28,12 +28,12 @@ namespace opossum {
  * data can no longer be generated. We decided that being able to generate tpcds data multiple times in for example a
  * hyriseConsole session is more important than fixing these small memory leaks (<1MB for 1GB of generated data).
  */
-class TpcdsTableGenerator final : public AbstractTableGenerator {
+class TPCDSTableGenerator final : public AbstractTableGenerator {
```
Comment (Bouncner) [inline src/benchmarklib/tpcds/tpcds_table_generator.hpp:31]: Best Renaming ever! EEeeverytime that made me nuts.

**Comment by Bouncner** [comment]
To understand the performance changes: what are the parts that benefit from the keys? Is there also something somewhere in the execution? I lost the overview.

**Comment by julianmenzler** [comment]
> To understand the performance changes: what are the parts that benefit from the keys? Is there also something somewhere in the execution? I've lost the overview.

Should be: 
1. DependentGroupByRule (using FDs)
2. ColumnPruningRule (using LQP unique constraints)

**Review by mrks — APPROVED** [review]
Neat :)

**Pull request merged**